lms/views/api/canvas/authorize.py

```diff
@@ -100,18 +101,13 @@ def oauth2_redirect(request):
     renderer="lms:templates/api/canvas/oauth2_redirect_error.html.jinja2",
 )
 def oauth2_redirect_error(request):
-    template_variables = {
-        "invalid_scope": request.params.get("error") == "invalid_scope",
-        "details": request.params.get("error_description"),
-        "scopes": FILES_SCOPES + SECTIONS_SCOPES,
-    }
-
-    if request.lti_user:
-        authorization_param = BearerTokenSchema(request).authorization_param(
-            request.lti_user
-        )
-        template_variables["authorize_url"] = request.route_url(
-            "canvas_api.authorize", _query=[("authorization", authorization_param)],
-        )
-
-    return template_variables
+    # In an exception view the `context` is usually the exception that occurred.
+    # We're going to cheat here and create a different object instead to serve
+    # as the `context` in templates.
+    context = FrontendAppResource(request)
+    context.js_config.enable_oauth_error_mode(
+        error_details=request.params.get("error_description"),
+        is_scope_invalid=request.params.get("error") == "invalid_scope",
+        requested_scopes=FILES_SCOPES + SECTIONS_SCOPES,
+    )
+    return {"context": context}
```

Comment:
Re. the comment above, feel free to suggest alternative approaches to creating the "context" like this.

Reply:
You first need to configure `FrontendAppResource` to be used as the context resource for non-error views for this view's route. Something like:

```diff
diff --git a/lms/routes.py b/lms/routes.py
index 3ac63406..24014b3f 100644
--- a/lms/routes.py
+++ b/lms/routes.py
@@ -25,7 +25,11 @@ def includeme(config):
         factory="lms.resources.LTILaunchResource",
     )
     config.add_route("canvas_oauth_callback", "/canvas_oauth_callback")
-    config.add_route("canvas_api.authorize", "/api/canvas/authorize")
+    config.add_route(
+        "canvas_api.authorize",
+        "/api/canvas/authorize",
+        factory="lms.resources.FrontendAppResource",
+    )
     config.add_route(
         "canvas_api.courses.files.list", "/api/canvas/courses/{course_id}/files"
     )
```

Now, even though the `context` argument that gets passed to exception views is the exception not the original context, the original context is still accessible as `request.context`:

```
diff --git a/lms/views/api/canvas/authorize.py b/lms/views/api/canvas/authorize.py
index 131403c9..30310425 100644
--- a/lms/views/api/canvas/authorize.py
+++ b/lms/views/api/canvas/authorize.py
@@ -101,13 +101,9 @@ def oauth2_redirect(request):
     renderer="lms:templates/api/canvas/oauth2_redirect_error.html.jinja2",
 )
 def oauth2_redirect_error(request):
-    # In an exception view the `context` is usually the exception that occurred.
-    # We're going to cheat here and create a different object instead to serve
-    # as the `context` in templates.
-    context = FrontendAppResource(request)
-    context.js_config.enable_oauth_error_mode(
+    request.context.js_config.enable_oauth_error_mode(
         error_details=request.params.get("error_description"),
         is_scope_invalid=request.params.get("error") == "invalid_scope",
         requested_scopes=FILES_SCOPES + SECTIONS_SCOPES,
     )
-    return {"context": context}
+    return {}
```